### DISCUSSION TITLE
My kingdom for circular arcs!

### DISCUSSION THREAD

**Discussion by Soothfast** [xournalpp/xournalpp#5560] [Ideas]
In making mathematical diagrams, I'm struck by how often it's necessary to draw portions of a circle, as opposed to a whole circle.  When conditions are right I might use a hack where I white-out a part of a circle and move it to where it's needed, but usually this is not feasible.  

On the other hand, the existing "draw spline" tool I find entirely useless.  It's too difficult and counterintuitive to implement.  More useful would be a "curve smoothing" tool:  I freehand a curve, maybe a little shaky or wavy in places, and "smooth curve" would render it as a professional-looking curve.  In fact such a smoothing tool would, for me, abrogate the need for a circular arc tool.

**Comment by rolandlo** [comment]
You can draw circular arcs using the compass

![grafik](https://github.com/xournalpp/xournalpp/assets/40485433/03c7fe8e-fa90-447e-a37d-ac72e22e9f7d)

There is currently only input stabilization (see `Input system` tab in menu `Edit > Preferences`), but a real "curve smoothening" tool is in the works as well: #2729

**Comment by Soothfast** [comment]
I'll play with that tool and see what's what.  I used to use another whiteboard application where a compass popped up, and it was only good for gauging the measures of angles.  I assumed this compass tool was the same.  Silly me?

**Reply by rolandlo** [reply]
In case you want to read up on how to use the geometry tools, here are [the docs](https://xournalpp.github.io/guide/tools/geometry/).

**Reply by Soothfast** [reply]
Oh, I figured it out.  Very nice.

The title of this thread promises a kingdom, but you called my bluff and I have no kingdom.  Do you guys accept donations? Your help alone has been worth the money.  :)

**Reply by rolandlo** [reply]
Thanks for your offer and the kind words. We don't accept donations, but we welcome donations to the projects we depend on. See https://github.com/xournalpp/xournalpp/discussions/3855#discussioncomment-2254937
